modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/bottomsheet/ModalBottomSheet.kt

```diff
@@ -0,0 +1,53 @@
+package au.com.shiftyjelly.pocketcasts.compose.bottomsheet
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ModalBottomSheetLayout
+import androidx.compose.material.ModalBottomSheetState
+import androidx.compose.material.ModalBottomSheetValue
+import androidx.compose.material.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterialApi::class)
```

Comment:
I'm using [ModalBottomSheetState](https://developer.android.com/reference/kotlin/androidx/compose/material/ModalBottomSheetState) which is experimental right now and I had to opt-in for `ExperimentalMaterialApi`. It made things really simple, I also tested it works on older Android versions (API level 23). But let me know if it's a deal breaker.

Reply:
I think that's fine. With compose still relatively young I don't think we should be too hesitant about using experimental apis when they make sense because a lot of fairly fundamental stuff still seems to be experimental.